### PR TITLE
Add generate-notebook to cli

### PR DESCRIPTION
**Pull request by wd15**
**Implement generate-notebook command**

List of changes:

 - Added pip install test without the working copy
 - Fix tests to work with nbval-lax as notebooks never give same results and difficult to write  regexs to deal with this
 - Remove weird `data_set` column from generated data frames
 - Fix bugs when converting between old and new schemas
   - One specific bug was with path names to data files
 - Allow cache clearing from cached data with the CLI
 - Enable rendering notebooks with both old and new style yaml files
 - Switch the simulation list from pfhub to pfhub-cli for now due to required changes as can't break the main website for now.
 - Added some test data permanently to the install

To review this:

 - Set up a new conda environment with Python 3.10 (isn't tested with 3.11 or 3.12)
 - pip install directly install from this branch
 - download a 1a.1 record with
   - `pfhub download meumapps_bm1a` for example (maybe choose some different records from this)
 - translate to old format
   - `mkdir new; pfhub convert ./meta.yaml --dest ./new`
 - build a benchmark notebook from the download
   - `pfhub render-notebook -r ./new/pfhub.yaml` (should work with the `meta.yaml` as well) 
   - Should be able to view the notebook
 - You should also be able to put together results list in YAML format and read a custom set of records with URLs, old and new style local YAMLs and build corresponding notebooks. 
 - Currently this only works for 1a.1. Subsequent PR will expand to other BMs

### CODE REVIEW

**Comment by guyer** [comment]
<details>

<summary>

```
$ mkdir old; pfhub convert-to-old ./meta.yaml --dest ./old
 ...
✗ Additional properties are not allowed ('_id', 'benchmark', 'data', 'date', 'metadata' were unexpected) in $
 ...
./meta.yaml is not valid
```

</summary>

```
WARNING:root:No namespace defined for URI: http://www.fairsharing.org/ontology/subject/SRAO_0000258
WARNING:root:No namespace defined for URI: https://en.wikipedia.org/wiki/Commit_(version_control)
WARNING:root:No namespace defined for URI: https://schema.org/identifier
WARNING:root:No namespace defined for URI: https://schema.org/contributor
WARNING:root:No namespace defined for URI: http://purl.obolibrary.org/obo/NCIT_C64194
WARNING:root:No namespace defined for URI: https://schema.org/dateCreated
WARNING:root:No namespace defined for URI: https://schema.org/downloadUrl
WARNING:root:No namespace defined for URI: https://schema.org/email
WARNING:root:No namespace defined for URI: https://schema.org/Number
WARNING:root:No namespace defined for URI: https://schema.org/name
WARNING:root:No namespace defined for URI: https://schema.org/encodingFormat
WARNING:root:No namespace defined for URI: https://schema.org/SoftwareApplication
WARNING:root:No namespace defined for URI: https://schema.org/member
WARNING:root:No namespace defined for URI: https://schema.org/identifier
WARNING:root:No namespace defined for URI: https://schema.org/name
WARNING:root:No namespace defined for URI: http://purl.obolibrary.org/obo/NCIT_C18132
WARNING:root:No namespace defined for URI: https://schema.org/url
WARNING:root:No namespace defined for URI: https://schema.org/url
WARNING:root:No namespace defined for URI: https://schema.org/abstract
WARNING:root:No namespace defined for URI: https://schema.org/Number
WARNING:root:No namespace defined for URI: https://schema.org/url
WARNING:root:No namespace defined for URI: https://schema.org/softwareVersion
WARNING:root:No namespace defined for URI: https://schema.org/softwareRequirements
WARNING:root:No namespace defined for URI: https://schema.org/Person
WARNING:root:No namespace defined for URI: https://schema.org/DigitalDocument
WARNING:root:No namespace defined for URI: https://schema.org/Dataset
WARNING:root:No namespace defined for URI: https://schema.org/SoftwareApplication
WARNING:root:No namespace defined for URI: https://schema.org/SoftwareSourceCode
WARNING:root:No namespace defined for URI: https://en.wikipedia.org/wiki/Comma-separated_values
✗ Additional properties are not allowed ('_id', 'benchmark', 'data', 'date', 'metadata' were unexpected) in $
✗ 'id' is a required property in $
✗ 'benchmark_problem' is a required property in $
✗ 'contributors' is a required property in $
✗ 'date_created' is a required property in $
✗ 'implementation' is a required property in $
✗ 'results' is a required property in $
✗ 'schema' is a required property in $
✗ 'summary' is a required property in $
./meta.yaml is not valid
```

</details>

**Comment by guyer** [comment]
```
$ pfhub render-benchmark -r ./meta.yaml
Usage: pfhub [OPTIONS] COMMAND [ARGS]...
Try 'pfhub --help' for help.

Error: No such command 'render-benchmark'.
```

Oh Lord, that should be `render-notebook`, not `render-benchmark`.

File: pfhub/func.py [hunk 1]
```diff
@@ -375,21 +414,25 @@ def compact(items):
...
     Args:
       keys: columns of each data item
       data: the data block with the given columns
+      base_path: base path to use in case of relative links
 
     Returns:
       The data columns in a pandas DataFrame
 
     """
+    add_base_path = lambda x: x if isabs(x) else os.path.join(base_path, x)
```
Comment (guyer) [inline pfhub/func.py:431]: Consider using [pathlib](https://docs.python.org/3/library/pathlib.html)

**Comment by wd15** [comment]
> ```
> $ mkdir old; pfhub convert-to-old ./meta.yaml --dest ./old
>  ...
> ✗ Additional properties are not allowed ('_id', 'benchmark', 'data', 'date', 'metadata' were unexpected) in $
>  ...
> ./meta.yaml is not valid
> ```
> 
> ```
> WARNING:root:No namespace defined for URI: http://www.fairsharing.org/ontology/subject/SRAO_0000258
> WARNING:root:No namespace defined for URI: https://en.wikipedia.org/wiki/Commit_(version_control)
> WARNING:root:No namespace defined for URI: https://schema.org/identifier
> WARNING:root:No namespace defined for URI: https://schema.org/contributor
> WARNING:root:No namespace defined for URI: http://purl.obolibrary.org/obo/NCIT_C64194
> WARNING:root:No namespace defined for URI: https://schema.org/dateCreated
> WARNING:root:No namespace defined for URI: https://schema.org/downloadUrl
> WARNING:root:No namespace defined for URI: https://schema.org/email
> WARNING:root:No namespace defined for URI: https://schema.org/Number
> WARNING:root:No namespace defined for URI: https://schema.org/name
> WARNING:root:No namespace defined for URI: https://schema.org/encodingFormat
> WARNING:root:No namespace defined for URI: https://schema.org/SoftwareApplication
> WARNING:root:No namespace defined for URI: https://schema.org/member
> WARNING:root:No namespace defined for URI: https://schema.org/identifier
> WARNING:root:No namespace defined for URI: https://schema.org/name
> WARNING:root:No namespace defined for URI: http://purl.obolibrary.org/obo/NCIT_C18132
> WARNING:root:No namespace defined for URI: https://schema.org/url
> WARNING:root:No namespace defined for URI: https://schema.org/url
> WARNING:root:No namespace defined for URI: https://schema.org/abstract
> WARNING:root:No namespace defined for URI: https://schema.org/Number
> WARNING:root:No namespace defined for URI: https://schema.org/url
> WARNING:root:No namespace defined for URI: https://schema.org/softwareVersion
> WARNING:root:No namespace defined for URI: https://schema.org/softwareRequirements
> WARNING:root:No namespace defined for URI: https://schema.org/Person
> WARNING:root:No namespace defined for URI: https://schema.org/DigitalDocument
> WARNING:root:No namespace defined for URI: https://schema.org/Dataset
> WARNING:root:No namespace defined for URI: https://schema.org/SoftwareApplication
> WARNING:root:No namespace defined for URI: https://schema.org/SoftwareSourceCode
> WARNING:root:No namespace defined for URI: https://en.wikipedia.org/wiki/Comma-separated_values
> ✗ Additional properties are not allowed ('_id', 'benchmark', 'data', 'date', 'metadata' were unexpected) in $
> ✗ 'id' is a required property in $
> ✗ 'benchmark_problem' is a required property in $
> ✗ 'contributors' is a required property in $
> ✗ 'date_created' is a required property in $
> ✗ 'implementation' is a required property in $
> ✗ 'results' is a required property in $
> ✗ 'schema' is a required property in $
> ✗ 'summary' is a required property in $
> ./meta.yaml is not valid
> ```

Sorry, I messed up here. It should have been just the `convert` (not `convert-to-old`) command so

      $ pfhub download meumapps_bm1a
      $ pfhub convert meta.yaml --dest=./old
   
That should work.

File: pfhub/func.py (hunk 1 above)
Comment (wd15) [reply pfhub/func.py:431]: `isabs` checks if it's a URL as well.

Comment (guyer) [reply pfhub/func.py:431]: Sorry, I meant for the `os.path` bits

Comment (wd15) [reply pfhub/func.py:431]: Ok, there are plenty of other places where `os.path.join` is used and some places where the pathlib paradigm is used. I should work on this as a separate PR so I've made an [issue](#16) for that.

**Review by guyer — APPROVED** [review]

**Pull request merged**